src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ProductController.php

```diff
@@ -288,14 +291,25 @@ public function lightListAction(ProductFilters $filters, Request $request): Resp
      *
      * @return RedirectResponse
      */
-    public function previewAction(int $productId): RedirectResponse
+    public function previewAction(int $productId, ?int $shopId): RedirectResponse
     {
-        /** @var bool $isEnabled */
-        $isEnabled = $this->getQueryBus()->handle(new GetProductIsEnabled((int) $productId));
+        $shopConstraint = !empty($shopId) ? ShopConstraint::shop($shopId) : ShopConstraint::allShops();
+        /** @var ProductForEditing $productForEditing */
+        $productForEditing = $this->getQueryBus()->handle(new GetProductForEditing(
+            $productId,
+            $shopConstraint,
+            $this->getContextLangId()
+        ));
+
+        if (null === $shopId) {
+            /** @var ProductMultiShopRepository $productRepository */
+            $productRepository = $this->get('prestashop.adapter.product.repository.product_multi_shop_repository');
```

Comment:
Maybe we should create the FQCN version of this service here and use it here. This way it would be easier to remove old service definitions.

Reply:
I started doing it at first, but it was only partially replaced If I want to modify everything properly it will increase the PR's size too much
So I changed my mind and I think we should have a dedicated PR that modify all the services related to product page in one go

Reply:
The less we have occurrences to change the better it is.
As far as I now since we decided to adopt FQCN nobody has spent time to make the necessary about old definition (create FQCN version with public false and aliases them with old name ...) IN A DEDICATED PR. We likely are going to do it, I'm sure, but it's going to be a hard task.
This occurence is in the scope of your PR. Can't we use FQCN now and change the remaining snake_case calls in a dedicated PR?